python-frontend/src/main/java/org/sonar/python/types/v2/FunctionType.java

```diff
@@ -121,4 +124,11 @@ public boolean hasVariadicParameter() {
   public PythonType owner() {
     return owner;
   }
+
+  public void resolveLazyReturnType(PythonType pythonType) {
+    if (!(returnType instanceof LazyType)) {
```

Comment:
same here, possibly we may want to move this check to utils

Reply:
I guess this case is a bit different since `resolveLazyReturnType` is essentially a setter that can be triggered only if the current return type is a lazy type. I still want to prevent using it if the current return type is not a lazy type.